### ISSUE TITLE
Create new directory if not exist when doing file renaming

### ISSUE THREAD

**Issue by amosbird** [Fuco1/dired-hacks#115]
Does this feature already exist in dired-hacks?

**Comment by Fuco1** [comment]
I don't think so, but I'm not sure about the context here.  Do you mean you would rename a file `foo` to `bar/foo` and we should therefore create the `bar` directory first?

I think that would be pretty cool in writable dired mode, I pretty much never rename files any other way.

**Comment by dieggsy** [comment]
@Fuco1 that actually does work in writable dired mode, I think.

**Comment by amosbird** [comment]
@Fuco1 yes, that's what I want to achieve.

@dieggsy Hmm, how does it work? I don't follow.

**Comment by Fuco1** [comment]
@dieggsy I get an error when I try that in wdired, maybe I have old version or something (still using 25.1)

@amosbird I think that's doable with some hooks/advice.  Might look into it some day.

**Comment by dieggsy** [comment]
@Fuco1 Ah, yeah I'm on a pretty recent git build but did some digging and the variable is `wdired-create-parent-directories` which is emacs >=26.1.

@amosbird well apparently it was introduced in emacs 26.1 but I can literally just enter wdired mode, take a file `foo`, type `bar/baz/foo` in its place and the directories `bar` and `baz` are just automatically created.
